From d258c78f06b11d4d557fc5c3861c9a142b08c45a Mon Sep 17 00:00:00 2001
From: wilson <wilson@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 6 Jun 2019 23:18:48 +0000
Subject: [PATCH 04/14] RISC-V: Move STARTFILE_PREFIX_SPEC into target OS
 files.

	gcc/
	PR target/89955
	* config/riscv/riscv.h (STARTFILE_PREFIX_SPEC): Deleted.
	* config/riscv/freebsd.h (STARTFILE_PREFIX_SPEC): Added.
	* config/riscv/linux.h (STARTFILE_PREFIX_SPEC): Added.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@272021 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog              | 7 +++++++
 gcc/config/riscv/freebsd.h | 6 ++++++
 gcc/config/riscv/linux.h   | 6 ++++++
 gcc/config/riscv/riscv.h   | 6 ------
 4 files changed, 19 insertions(+), 6 deletions(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index a8b782a79a0..0d1f7561d57 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,10 @@
+2019-06-06  Jim Wilson  <jimw@sifive.com>
+
+	PR target/89955
+	* config/riscv/riscv.h (STARTFILE_PREFIX_SPEC): Deleted.
+	* config/riscv/freebsd.h (STARTFILE_PREFIX_SPEC): Added.
+	* config/riscv/linux.h (STARTFILE_PREFIX_SPEC): Added.
+
 2019-04-30  Andrew Waterman  <andrew@sifive.com>
 	    Jim Wilson  <jimw@sifive.com>
 
diff --git a/gcc/config/riscv/freebsd.h b/gcc/config/riscv/freebsd.h
index 13d04ccbb47..bc516628285 100644
--- a/gcc/config/riscv/freebsd.h
+++ b/gcc/config/riscv/freebsd.h
@@ -52,3 +52,9 @@ along with GCC; see the file COPYING3.  If not see
         %{rdynamic:-export-dynamic}				\
         -dynamic-linker " FBSD_DYNAMIC_LINKER "}		\
         %{static:-static}}"
+
+#define STARTFILE_PREFIX_SPEC 			\
+   "/lib" XLEN_SPEC "/" ABI_SPEC "/ "		\
+   "/usr/lib" XLEN_SPEC "/" ABI_SPEC "/ "	\
+   "/lib/ "					\
+   "/usr/lib/ "
diff --git a/gcc/config/riscv/linux.h b/gcc/config/riscv/linux.h
index 58dd18b89f3..07ce80a847c 100644
--- a/gcc/config/riscv/linux.h
+++ b/gcc/config/riscv/linux.h
@@ -68,3 +68,9 @@ along with GCC; see the file COPYING3.  If not see
     %{static:-static}}"
 
 #define TARGET_ASM_FILE_END file_end_indicate_exec_stack
+
+#define STARTFILE_PREFIX_SPEC 			\
+   "/lib" XLEN_SPEC "/" ABI_SPEC "/ "		\
+   "/usr/lib" XLEN_SPEC "/" ABI_SPEC "/ "	\
+   "/lib/ "					\
+   "/usr/lib/ "
diff --git a/gcc/config/riscv/riscv.h b/gcc/config/riscv/riscv.h
index 4edd2a60194..8856cee599e 100644
--- a/gcc/config/riscv/riscv.h
+++ b/gcc/config/riscv/riscv.h
@@ -893,12 +893,6 @@ extern unsigned riscv_stack_boundary;
   "%{mabi=lp64f:lp64f}" \
   "%{mabi=lp64d:lp64d}" \
 
-#define STARTFILE_PREFIX_SPEC 			\
-   "/lib" XLEN_SPEC "/" ABI_SPEC "/ "		\
-   "/usr/lib" XLEN_SPEC "/" ABI_SPEC "/ "	\
-   "/lib/ "					\
-   "/usr/lib/ "
-
 /* ISA constants needed for code generation.  */
 #define OPCODE_LW    0x2003
 #define OPCODE_LD    0x3003
-- 
2.20.1

